src/System.Device.Gpio/System/Device/Gpio/Drivers/InterruptSysFsDriver.cs

```diff
@@ -23,6 +23,11 @@ public InterruptSysFsDriver(GpioDriver gpioDriver)
                 throw new PlatformNotSupportedException($"{GetType().Name} is only supported on Linux/Unix.");
             }
 
+            if (gpioDriver is null)
```

Comment:
NIT: Do you actually need this check? given gpioDriver is marked as non-nullable I think we can get rid of this, specially since we only call this internally

Reply:
You are right. The constructor value cannot be null with NRTs enabled.

What happens if NRTs is not enabled, and the null is passed? Won't this code nullref?

```csharp
        protected internal override PinValue Read(int pinNumber)
        {
            return _gpioDriver.Read(pinNumber);
        }
```

Reply:
correct, except that this is an internal type so it is actually only called from within the library itself, which we are already ensuring we won't pass null. More specifically, there is only one place where we instantiate this driver, which is from a RaspberryPi3LinuxDriver, and in there we pass `this` as the driver which of course won't be null:

https://github.com/dotnet/iot/blob/be64b3e55e64401aae36004e9722001e928e387f/src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs#L491